pkg/metadata/inventories_collector.go

```diff
@@ -59,9 +59,14 @@ func (c inventoriesCollector) Init() error {
 
 // SetupInventoriesExpvar init the expvar function for inventories
 func SetupInventoriesExpvar(ac inventories.AutoConfigInterface, coll inventories.CollectorInterface) {
+	if !config.Datadog.GetBool("enable_metadata_collection") {
+		log.Debugf("Metadata collection disabled: inventories payload will not be exposed to expvar")
+		return
+	}
```

Comment:
If I'm not mistaken, this is only directly called from the CLI check command which exits early if `enable_metadata_collection == true` and `SetupInventories` which won't publish the data to the backend and is gated with `inventories_enabled` config option. Do we need the special case for this?

Reply:
Since this function is part of the public API of the package I think it should.

Reply:
Fair. Would it then make sense to move both `inventories_enabled` and `enable_metadata_collection` config checks both into this package then or move both config checks to their callers? They both are gating this logic and it's a bit odd that one config item is evaluated in the callers and one is evaluated here function-level. Not a blocker but definitely inconsistent between the two.